scapy/fields.py

```diff
@@ -489,7 +489,7 @@ def getfield(self, pkt, s):
             lst.append(p)
         return remain+ret,lst
     def addfield(self, pkt, s, val):
-        return s+"".join(map(str, val))
+        return s + "".join([str(v) for v in val])
```

Comment:
Same for every other instance of `.join([... for ... in ...])`: `.join(... for ... in ...)` will do the same without creating a list.